### ISSUE TITLE
Documentation is very poor, bis!

### ISSUE THREAD

**Issue by haterecoil** [sendinblue/APIv3-php-library#70]
Dear maintainers,

What the hell? The PHP library is both poorly designed and poorly documented.

PS: this is a product we are paying for. As of 2019 a SDK is the bare minimum you could provide. FYI here is the pricing page of SendinBlue : https://account.sendinblue.com/pricing/

**Comment by COil** [comment]
Sendgrid PHP API qulity was also low some time ago, it's better now. 

But +1 for the doc. For example for the `sendtransacemail` function doc : 

    $sendSmtpEmail = new \SendinBlue\Client\Model\SendSmtpEmail(); // \SendinBlue\Client\Model\SendSmtpEmail | Values to send a transactional email
 
    try {
        $result = $apiInstance->sendTransacEmail($sendSmtpEmail);
        print_r($result);
    } catch (Exception $e) {
        echo 'Exception when calling SMTPApi->sendTransacEmail: ', $e->getMessage(), PHP_EOL;
    }

It's not real code as SendSmtpEmail is created without argument. Such a code leads to an error 400 : 

    Client error: `POST https://api.sendinblue.com/v3/smtp/email` resulted in a `400 Bad Request` response:
    {"code":"missing_parameter","message":"No input in body"}

I had to figure out how to pass the to parameter as the documentation is no correct. In fact "Models" sometimes seem not to have to be used ! :thinking:

**Comment by vcopleutre** [comment]
Hi @COil,

How did you pass the parameters (i'm fighting with json types that are refused by Sendinblue).

My current code : 
`$params[] = json_encode(array(
            'TITLE' => 'testtitle',
            'TITLE2' => 'testtitle'
        ));
        $sendSmtpEmail->setParams($params);`

It doesn't work. Sendinblue log write : 
`"build template args: decode: json: cannot unmarshal array into Go struct field httpRequestTemplateArgs.params of type map[string]interface {}\n"`

I'm stuck with it and i would love a little help. Btw I agree with the fact that documentation is very poor... 

Thank you :-)

**Comment by COil** [comment]
$container = [];
        $email = ['name' => 'Loïc Vernet', 'email' => 'coil@bar.com'];
        $container['textContent'] = 'Hello world !';
        $container['subject'] = 'SendinBlue test via V3 PHP API';
        $container['sender'] = new SendSmtpEmailSender($email);
        $container['replyTo'] = new SendSmtpEmailReplyTo($email);
        $container['to'] = [$email];
        $container['tags'] = ['test'];
        $sendSmtpEmail = new SendSmtpEmail($container);

        return $this->smtpApi->sendTransacEmail($sendSmtpEmail);

**Comment by vcopleutre** [comment]
Thank you very much !

Unfortunately I dont succeed in using my params in my template even without json_encoding.
`%VARIABLE%` is not replaced by my params set by `$container['params'] = ['VARIABLE' => 'test variable']`

I'm trying to reach the tech support

**Comment by vcopleutre** [comment]
AMEN.
For those who want to pass parameters in templates. The correct syntax is : 

1. Pass parameters in the form of array in PHP : ['VARIABLE' => 'test variable']

**Comment by boryn** [comment]
Hi @vcopleutre! Your comment is a life-saver! 

Did you maybe find out how to use `{{ contact.FIRSTNAME }}` "transactionally"? I mean I'd like to pass the user name on the fly from my database, without a necessity of creating a contact in Sendinblue. Is there any special way to pass the `contact` data or I just should use {{ params.FIRSTNAME }}?

**Comment by boryn** [comment]
I went with the solution with `{{ params.FIRSTNAME }}` as I trust more data in my database than something automatically created by Sendinblue.

Another important thing: you cannot use hyphens '-' in the variable names. So `{{ params.FIRST_NAME }}` will work properly, but `{{ params.FIRST-NAME }}` will fail.

For those working in Laravel with Sendinblue, I have extended the https://github.com/agence-webup/laravel-sendinblue library for using the Sendinblue templates and transactional parameters (https://github.com/agence-webup/laravel-sendinblue/pull/43).

**Comment by tbba** [comment]
Yes, the poor documentation (no kitchen sink examples) is why we  started evaluating alternatives already.

**Comment by preetishishodia7** [comment]
Hi, 
Can you please tell, do you still face issues with the documentation part in order to use it, if yes can you please share where you are facing issues? So that we can check and improve it, as its an old thread we have changed a lot in our documentation from then.
Thanks!
